ddtrace/internal/writer.py

```diff
@@ -120,40 +102,57 @@ def flush_queue(self):
         # Dump statistics
         # NOTE: Do not use the buffering of dogstatsd as it's not thread-safe
         # https://github.com/DataDog/datadogpy/issues/439
-        if send_stats:
+        if self._send_stats:
             # Statistics about the queue length, size and number of spans
-            self.dogstatsd.gauge('datadog.tracer.queue.max_length', self._trace_queue.maxsize)
-            self.dogstatsd.gauge('datadog.tracer.queue.length', traces_queue_length)
-            self.dogstatsd.gauge('datadog.tracer.queue.spans', traces_queue_spans)
-
-            # Statistics about the rate at which spans are inserted in the queue
-            dropped, enqueued, enqueued_lengths = self._trace_queue.reset_stats()
-            self.dogstatsd.increment('datadog.tracer.queue.dropped', dropped)
-            self.dogstatsd.increment('datadog.tracer.queue.accepted', enqueued)
-            self.dogstatsd.increment('datadog.tracer.queue.accepted_lengths', enqueued_lengths)
+            self.dogstatsd.increment('datadog.tracer.flushes')
+            self.dogstatsd.histogram('datadog.tracer.flush.traces', traces_queue_length)
+            self.dogstatsd.histogram('datadog.tracer.flush.spans', traces_queue_spans)
 
             # Statistics about the filtering
-            self.dogstatsd.increment('datadog.tracer.traces.filtered', traces_filtered)
+            self.dogstatsd.histogram('datadog.tracer.flush.traces_filtered', traces_filtered)
 
             # Statistics about API
-            self.dogstatsd.increment('datadog.tracer.api.requests', len(traces_responses))
-            self.dogstatsd.increment('datadog.tracer.api.errors',
+            self.dogstatsd.histogram('datadog.tracer.api.requests', len(traces_responses))
+            self.dogstatsd.histogram('datadog.tracer.api.errors',
                                      len(list(t for t in traces_responses
                                               if isinstance(t, Exception))))
             for status, grouped_responses in itertools.groupby(
                     sorted((t for t in traces_responses if not isinstance(t, Exception)),
                            key=lambda r: r.status),
                     key=lambda r: r.status):
-                self.dogstatsd.increment('datadog.tracer.api.responses',
+                self.dogstatsd.histogram('datadog.tracer.api.responses',
                                          len(list(grouped_responses)),
                                          tags=['status:%d' % status])
 
             # Statistics about the writer thread
             if hasattr(time, 'thread_time_ns'):
                 self.dogstatsd.increment('datadog.tracer.writer.cpu_time', time.thread_time_ns())
```

Comment:
didn't you want to fix this too? I think the increment for this was not correct, might be better as a gauge

Reply:
I didn't hit submit on the other PR I have ready, just wait like... 2 minutes :)